### PR TITLE
[DI] Add samples and update README

### PR DESCRIPTION
**Pull request by YalinLi0312**
This PR includes:
* New samples for how to convert AnalyzeResult to a JSON serializable dict, it's asked in [this](https://github.com/Azure/azure-sdk-for-python/issues/34027) customer issue.
* New samples for analyzing document as base64 source, it's asked in [this](https://github.com/Azure/azure-sdk-for-python/issues/33901) customer issue. 
Update: we have renamed `base64_source` to `bytes_source` and added samples [sample_analyze_invoices_from_bytes_source.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_invoices_from_bytes_source.py) and [sample_analyze_invoices_from_bytes_source_async.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_analyze_invoices_from_bytes_source_async.py). So removed base64 samples after first round review.
* Disclaim available regions for API 2024-02-29-preview, it's asked in [this](https://github.com/Azure/azure-sdk-for-python/issues/33312) customer issue.
* Added links for some DI docs which were not available when shipping first release.

### CODE REVIEW

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    result: AnalyzeResult = poller.result()
+
+    # convert the received model to a dictionary
+    analyze_result_dict = result.as_dict()
+
+    # save the dictionary as JSON content in a JSON file, use the AzureJSONEncoder
+    # to help make types, such as dates, JSON serializable
+    # NOTE: AzureJSONEncoder is only available with azure.core>=1.18.0.
+    with open("data.json", "w") as output_file:
+        json.dump(analyze_result_dict, output_file, cls=AzureJSONEncoder)
```
Comment (lmazuel) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py:61]: You shouldn't need this line, `as_dict` should give you something that can be passed directly to `json.dumps`

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_invoices_as_base64_source.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    from azure.ai.documentintelligence import DocumentIntelligenceClient
+    from azure.ai.documentintelligence.models import AnalyzeResult, AnalyzeDocumentRequest
+
+    endpoint = os.environ["DOCUMENTINTELLIGENCE_ENDPOINT"]
+    key = os.environ["DOCUMENTINTELLIGENCE_API_KEY"]
+
+    document_intelligence_client = DocumentIntelligenceClient(endpoint=endpoint, credential=AzureKeyCredential(key))
+    with open(path_to_sample_documents, "rb") as f:
+        poller = document_intelligence_client.begin_analyze_document(
+            "prebuilt-invoice", AnalyzeDocumentRequest(base64_source=f.read()), locale="en-US"
```
Comment (lmazuel) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_invoices_as_base64_source.py:49]: You're opening a JPG. A JPEG is not base64 data, so I'm confused here?

Comment (lmazuel) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_invoices_as_base64_source.py:49]: Ok, I understand what is happening. The RestAPI needs based64 on the wire. However, the codegen is doing the base64 encoding on behalf of customers thwn they pass bytes.

The problem here, is that the parameter is infered from the RestAPI parameter, and therefore called "base64_source", giving the feeling to customer they can pass base64. But they can't, they need to pass bytes.

This parameter needs a rename (clientName), so customers understand they need to pass raw bytes.

This should be clarified in the sample.

CC @johanste

Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_invoices_as_base64_source.py:49]: Yes, it's not expected to pass base64 encoded content from customers, our SDK would encode the data automatically. I have requested other languages to confirm if they have the same behavior, and requested a name change in the same email thread.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py (hunk 1 above)
Comment (muizzhaque) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py]: I tried to pass as_dict directly to json.dumps, but the result is not what I would expect from a json file, no indentations and in general it is somewhat similar to how you would store a text file.

Where as json.dump has what a json file should like

Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py]: @muizzhaque  You should able to format the output with either `json.dumps` or `json.dump`, for example:
```python
with open("data.json", "w") as output_file:
    json.dump(analyze_result_dict, output_file, indent=4)
```
or
```python
json.dumps(analyze_result_dict, indent=4)
```

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+DESCRIPTION:
+    This sample demonstrates how to convert a model returned from an analyze operation
+    to a JSON serializable dictionary.
+
+USAGE:
+    python sample_convert_to_dict.py
+
+    Set the environment variables with your own values before running the sample:
+    1) DOCUMENTINTELLIGENCE_ENDPOINT - the endpoint to your Form Recognizer resource.
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py:20]: Nit: 
```suggestion
    1) DOCUMENTINTELLIGENCE_ENDPOINT - the endpoint to your Document Intelligence resource.
```

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    2) DOCUMENTINTELLIGENCE_API_KEY - your Form Recognizer API key
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py:21]:
```suggestion
    2) DOCUMENTINTELLIGENCE_API_KEY - your Document Intelligence API key
```

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_convert_to_dict_async.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+DESCRIPTION:
+    This sample demonstrates how to convert a model returned from an analyze operation
+    to a JSON serializable dictionary.
+
+USAGE:
+    python sample_convert_to_dict_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) DOCUMENTINTELLIGENCE_ENDPOINT - the endpoint to your Form Recognizer resource.
+    2) DOCUMENTINTELLIGENCE_API_KEY - your Form Recognizer API key
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_convert_to_dict_async.py:21]:
```suggestion
    1) DOCUMENTINTELLIGENCE_ENDPOINT - the endpoint to your Document Intelligence resource.
    2) DOCUMENTINTELLIGENCE_API_KEY - your Document Intelligence API key
```

**Review by catalinaperalta — APPROVED** [review]
Looks like this PR isnt changing anything related to the base64->bytes_source change mentioned in the description. The samples added lgtm!

**Comment by YalinLi0312** [comment]
> Looks like this PR isnt changing anything related to the base64->bytes_source change mentioned in the description. The samples added lgtm!

The base64 rename is already merged in another PR, so I reverted any related changes here. Thanks for your review!

**Pull request merged**